### DISCUSSION TITLE
[Interface] Color-Coded Interface (ala Heroes III)

### DISCUSSION THREAD

**Discussion by JCollins2048** [ihhub/fheroes2#8861] [Ideas]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

I've been watching a lot of *Heroes of Might and Magic III* videos lately.  Something I've noticed about the interface of that game is that parts of the user interface are color-coded.  For example, if the current player is Red, then the Resource Bar at the bottom is also red-colored, as are the various option buttons between the Heroes and Towns columns.  So, I got to thinking… wouldn't that a fun thing to add to *fheroes2*?

### Describe the possible solution

Basically, I've come up with three possible options on how to implement this:

1. Make all of the gems on either interface the same color as the player's hero.
This would be *incredibly* subtle, especially on the Evil interface, but it would allow players (and viewers) to more clearly differentiate players from one another at a glance.  This would require making the gems all the same color, then having that color change purely based on the player, which could be annoying — once again, especially in the case of the Evil Interface's glowing gems.

2. Make the Interface buttons the same color as the player's hero.
This would be a lot more overt and hard to miss.  The downside is that it can be kind of *ugly*, since everything surrounding the Interface would remain its usual colors.  This should be easier to do, in theory, but require the same amount of coding and graphic editing, I think.
 
3. Make a new Interface option which is a little closer to that of *Heroes III*.
And of course, *this* is the most complicated choice of them all as it would require making brand new resources (or grabbing them out of *Heroes III*…?) and take the most time to implement, in theory.

### Additional info

This might seem like a waste of time for some players, but I think this could be useful in Hotseat games and for "Let's Plays".  As I said, I've been watching a lot of *Heroes III* Let's Plays, and when I see a *Heroes II* one by the same player, I tend to get momentarily confused as to what they're looking at since they're scrolling all over the map and not necessarily focused on their own property.  Having some kind of colored indication for whoever's currently player could be helpful in these instances.

I've made mockups of what this could potentially look like, though it could probably be done better.
(In hindsight, I probably should've picked a hero color that wasn't already a major part of the Evil interface…)

### Relevant Screenshots
<img width="33%" src="https://github.com/ihhub/fheroes2/assets/102710555/e3cd0792-fa8b-4c58-98d1-66637fbf9449"> <img width="33%" src="https://github.com/ihhub/fheroes2/assets/102710555/e4e8c4bf-5699-4f39-8cdd-f9f05c3dd30b"><br>*The original "Good" interface vs. a mockup of a "Good" interface color-coded for Blue.  Note the gems and the control panel.*

<img width="33%" src="https://github.com/ihhub/fheroes2/assets/102710555/bed5ec48-cf9d-473b-b02e-46ee8cfa8a5a"> <img width="33%" src="https://github.com/ihhub/fheroes2/assets/102710555/3afcf3a7-86fd-4905-aa3a-d081b0a0f34f"><br>*Ditto for the "Evil" interface.*

**Comment by ihhub** [comment]
Moving this topic to Discussion section.
